keytree/src/lib.rs

```diff
@@ -50,6 +50,39 @@ impl Xprv {
             precompressed_pubkey: self.precompressed_pubkey,
         }
     }
+
+    /// Serializes this Xprv to a sequence of bytes.
+    pub fn to_bytes(&self) -> [u8; 64] {
+        let mut buf = [0u8; 64];
+        buf[..32].copy_from_slice(&self.scalar.to_bytes());
+        buf[32..].copy_from_slice(&self.dk);
+        buf
+    }
+
+    /// Decodes an Xprv from a 64-byte array, and fails if the provided array is not
+    /// exactly 64 bytes.
+    pub fn from_bytes(bytes: &[u8]) -> Option<Self> {
+        if bytes.len() != 64 {
+            return None;
+        }
+
+        let (scslice, dkslice) = bytes.split_at(32);
```

Comment:
sorry, i overlooked it previously. You probably don't need this explicit splitting since you can slice things up in place:

```rust
scalar_bytes.copy_from_slice(&bytes[..32]);
...
dk.copy_from_slice(&bytes[32..]);
```

And the same for xpub parsing.